### ISSUE TITLE
Example: Bayesian Neural Network, suggestion for improvement

### ISSUE THREAD

**Issue by paapu88** [pyro-ppl/numpyro#1236]
Dear All,
In example for Bayesian Neural Network
http://num.pyro.ai/en/stable/examples/bnn.html
neural nets are represented as matrix multiplication and bias term is left out.

I was pretty confused first because I could not get reproducible results...

In order to mimic "normal" fully connected layer one should replace
A@x by A@x + b (@ stands for matrix multiplication here)

example code as it is in http://num.pyro.ai/en/stable/examples/bnn.html:
```
def model(X, Y, D_H, D_Y=1):
    N, D_X = X.shape
    # sample first layer (we put unit normal priors on all weights)
    w1 = numpyro.sample("w1", dist.Normal(jnp.zeros((D_X, D_H)), jnp.ones((D_X, D_H))))
    assert w1.shape == (D_X, D_H)
    z1 = nonlin(jnp.matmul(X, w1))  # <= first layer of activations
```

improvement suggestion (with  A@x + b, and jax.relu activation instead of hand-made tanh in the example)):
```
def model(X, Y, D_H, D_Y=1):
    N, D_X = X.shape
    # sample first layer (we put unit normal priors on all weights)
    w1 = numpyro.sample("w1", dist.Normal(jnp.zeros((D_X, D_H)), jnp.ones((D_X, D_H))))
    b1 = numpyro.sample("b1", dist.Normal(jnp.zeros((D_H)), jnp.ones((D_H))))
    assert w1.shape == (D_X, D_H)
    z1 = jax.nn.relu(jnp.matmul(X, w1) + b1)  # <= first layer of activations
    .. etc ...
```

Also one could think about adding normalization which seems to be implemented in jax.nn :
https://jax.readthedocs.io/en/latest/_autosummary/jax.nn.normalize.html#jax.nn.normalize

Thanks for great code and documentation!!! 
terveisin, Markus

**Comment by martinjankowiak** [comment]
@paapu88 hi markus please feel free to submit a PR if you would like to see a bias term added. i made the choice to keep the architecture simple because that's all that's necessary for this simple dataset. note also that the inputs in the artificial dataset have a "all ones" component. as such there's effectively a bias term in the first layer anyway. if you want to add explicit bias terms to all layers, you'll need to change `get_data` to remove the all ones component which would then be redundant.

**Comment by paapu88** [comment]
Ok, I guess I could write another simple example using bayesian neural net. However, I'm novice on this, so it may take some time... still reading the great book "Statistical Rethinking: A Bayesian Course With..."

**Comment by OlaRonning** [comment]
@paapu88 feel free to take inspiration/code from [this](https://github.com/aleatory-science/numpyro/blob/feature/einstein-interface/examples/bayes_networks_stein.py#L41) bnn implementation.
